command/pr_lookup.go

```diff
@@ -10,28 +10,36 @@ import (
 	"github.com/cli/cli/context"
 	"github.com/cli/cli/git"
 	"github.com/cli/cli/internal/ghrepo"
+	"github.com/spf13/cobra"
 )
 
-func prFromArgs(ctx context.Context, apiClient *api.Client, repo ghrepo.Interface, args []string) (*api.PullRequest, error) {
+func prFromArgs(ctx context.Context, apiClient *api.Client, cmd *cobra.Command, args []string) (*api.PullRequest, ghrepo.Interface, error) {
+	repo, err := determineBaseRepo(apiClient, cmd, ctx)
```

Comment:
We could hold off invoking this in case a URL was passed as argument, since this triggers an API request that is not necessary as the base repo information will be read from URL.

Reply:
Did this in https://github.com/cli/cli/pull/1020/commits/f490a4914f5fad55fde647352aab2a9e2d04ec43 and works well. Thanks for the idea.